solidity/ecdsa/contracts/DKGValidator.sol

```diff
@@ -0,0 +1,251 @@
+                groupSize
+            ) {
+                return (false, "Corrupted misbehaved members indices");
+            }
+            for (uint256 i = 1; i < misbehavedMembersIndices.length; i++) {
+                if (
+                    misbehavedMembersIndices[i - 1] >=
+                    misbehavedMembersIndices[i]
+                ) {
+                    return (false, "Corrupted misbehaved members indices");
+                }
+            }
+        }
+
+        // Each signature needs to have a correct length and signatures need to
+        // be provided.
+        uint256 signaturesCount = result.signatures.length / signatureByteSize;
+        if (result.signatures.length == 0) {
+            return (false, "No signatures provided");
+        }
+        if (result.signatures.length % signatureByteSize != 0) {
+            return (false, "Malformed signatures array");
+        }
+
+        // We expect the same amount of signatures as the number of declared
+        // group member indices that signed the result.
+        uint256[] calldata signingMembersIndices = result.signingMembersIndices;
+        if (signaturesCount != signingMembersIndices.length) {
+            return (false, "Unexpected signatures count");
+        }
+        if (signaturesCount < groupThreshold) {
+            return (false, "Too few signatures");
+        }
+        if (signaturesCount > groupSize) {
+            return (false, "Too many signatures");
+        }
+
+        // Signing member indices needs to be unique, between [1,groupSize],
+        // and sorted in ascending order.
+        if (
+            signingMembersIndices[0] < 1 ||
+            signingMembersIndices[signingMembersIndices.length - 1] > groupSize
+        ) {
+            return (false, "Corrupted signing member indices");
+        }
+        for (uint256 i = 1; i < signingMembersIndices.length; i++) {
+            if (signingMembersIndices[i - 1] >= signingMembersIndices[i]) {
+                return (false, "Corrupted signing member indices");
+            }
+        }
+
+        return (true, "");
+    }
+
+    /// @notice Performs validation of group members as declared in DKG
+    ///         result against group members selected by the sortition pool.
+    /// @param seed seed used to start the DKG and select group members
+    /// @return true if group members matches; false otherwise
+    function validateGroupMembers(DKG.Result calldata result, uint256 seed)
+        public
+        view
+        returns (bool)
+    {
+        // Compute the actual group members hash by selecting actual members IDs
+        // based on seed used for current DKG execution.
+        bytes32 actualGroupMembersHash = keccak256(
+            abi.encodePacked(
+                sortitionPool.selectGroup(groupSize, bytes32(seed))
+            )
+        );
+
+        // TODO: check what is more efficient - computing hash or iterating
```

Comment:
Can we remove it as part of #2779 from the random-beacon's library and then we will just copy-paste it here? I want the `DKGValidator.sol` file to be a copy-paste with just a few crucial changes.

Reply:
The thing is that if we leave 64 group members in the random-beacon then it is more efficient gas wise to use the iterative approach implemented in [this PR](https://github.com/keep-network/keep-core/pull/2753). Only if we have 100members then gas on average will be the same for hashing vs. iterative, but hashing is more concise and cleaner imo. So I'd rather merge #2753, next #2779 but at this point, it will be already changed to an iterative approach, so if we want to leave hashing here, the copy-pasting 1:1 won't work.

Unless at some point we'll extract common libs to a separate repo like I mentioned in [here](https://github.com/keep-network/keep-core/pull/2775#pullrequestreview-852945168), then I guess we can temporarily leave the iterative approach in /ecdsa as well.